x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/text_expansion_callout_logic.ts

```diff
@@ -260,5 +261,11 @@ export const TextExpansionCalloutLogic = kea<
       () => [selectors.startTextExpansionModelStatus],
       (status: Status) => status !== Status.IDLE && status !== Status.ERROR,
     ],
+    isModelRunningSingleThreaded: [
+      () => [selectors.textExpansionModel],
+      (data: FetchTextExpansionModelResponse) =>
+        // Running single threaded if model has max 1 deployment on 1 node with 1 thread
+        data?.allocationCount * data?.nodeAllocationCount * data?.threadsPerAllocation <= 1,
```

Comment:
cc @droberts195 - as per Slack thread.

Reply:
It's probably best to use `target_allocation_count` instead of `allocation_count` here if the objective is to detect when a user chose single threaded rather than when a model is temporarily single threaded because it's only just starting up.

If I choose 10 allocations then there will still be a second or two during startup when only 1 of the 10 has started.

Reply:
@droberts195 `allocationCount` is the number of items in `trained_model_stats` for this model. So if we successfully start the model 2 times with different configurations, it will show 2.
Originally I didn't include this in the formula, but I thought if there were multiple deployments, that would make the model more balanced and highly available, and not count as single-threaded anymore. Is that inaccurate?

(`nodeAllocationCount` and `threadsPerAllocation` come directly from 1st deployment, i.e. `trained_model_stats[0]`)

Reply:
> I thought if there were multiple deployments, that would make the model more balanced and highly available

The reason for starting the model twice with different deployment IDs is if it's needed for 2 use cases, such as search and indexing. Starting it twice with different deployment IDs _won't_ make it more highly available for a single use case. For example, a model deployed separately for search and indexing but with only 1 allocation per deployment will go offline during rolling upgrades. It might go offline at different times for search and indexing, as different nodes are restarted, but won't be highly available for either.

In terms of whether something is single threaded, that really needs to be the deployment rather than the model. However, if that's too big a change for this stage of the 8.8 development process then what you have is probably OK for a first release, and maybe the view can be changed for a future release.

Reply:
Got it. I updated the condition to check if `target_allocation_count *  threads_per_allocation <= 1`.